File: LinkedList/LinkedList.js
class Node {
    constructor(data) {
        this.data = data;
        this.next = null;
    }

    // get the next node
    getNext() {
        return this.next;
    }

    // add the next node
    addNext(node) {
        this.next = node;
    }
}

function main() {
    // Creating and linking nodes together in one step
    const firstNode = new Node(3);
    firstNode.addNext(new Node(5));
    firstNode.getNext().addNext(new Node(13));
    firstNode.getNext().getNext().addNext(new Node(2));

    // Printing linked list
    let currentNode = firstNode;
    while (currentNode !== null) {
        process.stdout.write(currentNode.data + " -> ");
        currentNode = currentNode.getNext();
    }
    console.log("null");
}

main();

File: LinkedList/LinkedList.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

    def setNext(self, next_node):
        self.next = next_node

    def getNext(self):
        return self.next

# Creating nodes
node1 = Node(3)
node2 = Node(5)
node3 = Node(13)
node4 = Node(2)

# Linking nodes using setNext()
node1.setNext(node2)
node2.setNext(node3)
node3.setNext(node4)

# Traversing the linked list and printing data
currentNode = node1
while currentNode:
    print(currentNode.data, end=" -> ")
    currentNode = currentNode.getNext()
print("null")
